undertow/src/main/java/org/wildfly/extension/undertow/ApplicationSecurityDomainDefinition.java

```diff
@@ -285,7 +285,10 @@ protected void performRuntime(OperationContext context, ModelNode operation, Res
                 ModelNode ssoModel = resource.getChild(UndertowExtension.PATH_SSO).getModel();
 
                 String cookieName = SingleSignOnDefinition.Attribute.COOKIE_NAME.resolveModelAttribute(context, ssoModel).asString();
-                String domain = SingleSignOnDefinition.Attribute.DOMAIN.resolveModelAttribute(context, ssoModel).asString();
+                String domain = null;
```

Comment:
@hokuda It would be less verbose using the new-ish asStringOrNull() method. e.g:
String domain = SingleSignOnDefinition.Attribute.DOMAIN.resolveModelAttribute(context, ssoModel).asStringOrNull()

@stuartwdouglas apart from that please approve